src/darwin/Framework/CHIP/CHIPPersistentStorageDelegateBridge.mm

```diff
@@ -112,43 +114,52 @@
     return error;
 }
 
-void CHIPPersistentStorageDelegateBridge::AsyncSetKeyValue(const char * key, const char * value)
+CHIP_ERROR CHIPPersistentStorageDelegateBridge::SyncSetKeyValue(const char * key, const void * value, uint16_t size)
 {
+    std::string base64Value = StringToBase64(std::string(static_cast<const char *>(value), size));
+
     NSString * keyString = [NSString stringWithUTF8String:key];
-    NSString * valueString = [NSString stringWithUTF8String:value];
-    dispatch_async(mWorkQueue, ^{
-        NSLog(@"PersistentStorageDelegate Set Key %@, Value %@", keyString, valueString);
+    NSString * valueString = [NSString stringWithUTF8String:base64Value.c_str()];
+
+    dispatch_sync(mWorkQueue, ^{
+        NSLog(@"PersistentStorageDelegate Set Key %@", keyString);
 
         id<CHIPPersistentStorageDelegate> strongDelegate = mDelegate;
         if (strongDelegate && mQueue) {
-            dispatch_async(mQueue, ^{
-                [strongDelegate CHIPSetKeyValue:keyString value:valueString handler:mSetStatusHandler];
+            dispatch_sync(mQueue, ^{
+                [strongDelegate CHIPSetKeyValue:keyString value:valueString];
             });
         } else {
             [mDefaultPersistentStorage setObject:valueString forKey:keyString];
-            if (mSetStatusHandler) {
-                mSetStatusHandler(keyString, [CHIPError errorForCHIPErrorCode:0]);
-            }
         }
     });
+
+    // TODO: ideally the error from the dispatch should be returned
+    // however we expect to replace the storage delegate with KVS so for now
+    // we return no error (return used to be void due to async dispatch anyway)
+
+    return CHIP_NO_ERROR;
 }
 
-void CHIPPersistentStorageDelegateBridge::AsyncDeleteKeyValue(const char * key)
+CHIP_ERROR CHIPPersistentStorageDelegateBridge::SyncDeleteKeyValue(const char * key)
 {
     NSString * keyString = [NSString stringWithUTF8String:key];
-    dispatch_async(mWorkQueue, ^{
+    dispatch_sync(mWorkQueue, ^{
         NSLog(@"PersistentStorageDelegate Delete Key: %@", keyString);
 
         id<CHIPPersistentStorageDelegate> strongDelegate = mDelegate;
         if (strongDelegate && mQueue) {
-            dispatch_async(mQueue, ^{
-                [strongDelegate CHIPDeleteKeyValue:keyString handler:mDeleteStatusHandler];
+            dispatch_sync(mQueue, ^{
```

Comment:
Darwin doesn't promote sync access, if you want to make it sync - you create a queue, and dispatch sync to that queue for the completion.